userspace/falco/outputs_http.cpp

```diff
@@ -47,62 +56,83 @@ void falco::outputs::output_http::output(const message *msg)
 			{
 				unquotedUrl = libsinsp::filter::unescape_str(unquotedUrl);
 			}
-			res = curl_easy_setopt(curl, CURLOPT_URL, unquotedUrl.c_str());
-		}
-
-		if(res == CURLE_OK)
-		{
-			res = curl_easy_setopt(curl, CURLOPT_POSTFIELDS, msg->msg.c_str());
+			res = curl_easy_setopt(m_curl, CURLOPT_URL, unquotedUrl.c_str());
 		}
 
 		if(res == CURLE_OK)
 		{
-			res = curl_easy_setopt(curl, CURLOPT_USERAGENT, m_oc.options["user_agent"].c_str());
+			res = curl_easy_setopt(m_curl, CURLOPT_USERAGENT, m_oc.options["user_agent"].c_str());
 		}
 
 		if(res == CURLE_OK)
 		{
-		   res = curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, -1L);
+			res = curl_easy_setopt(m_curl, CURLOPT_POSTFIELDSIZE, -1L);
 		}
 
 		if(res == CURLE_OK)
 		{
 			if(m_oc.options["insecure"] == std::string("true"))
 			{
-				res = curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER, 0L);
+				res = curl_easy_setopt(m_curl, CURLOPT_SSL_VERIFYPEER, 0L);
 
 				if(res == CURLE_OK)
 				{
-					res = curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
+					res = curl_easy_setopt(m_curl, CURLOPT_SSL_VERIFYHOST, 0L);
 				}
 			}
 		}
 
 		if(res == CURLE_OK)
 		{
-			if (!m_oc.options["ca_cert"].empty())
+			if(!m_oc.options["ca_cert"].empty())
+			{
+				res = curl_easy_setopt(m_curl, CURLOPT_CAINFO, m_oc.options["ca_cert"].c_str());
+			}
+			else if(!m_oc.options["ca_bundle"].empty())
 			{
-				res = curl_easy_setopt(curl, CURLOPT_CAINFO, m_oc.options["ca_cert"].c_str());
-			}else if(!m_oc.options["ca_bundle"].empty())
+				res = curl_easy_setopt(m_curl, CURLOPT_CAINFO, m_oc.options["ca_bundle"].c_str());
+			}
+			else
 			{
-				res = curl_easy_setopt(curl, CURLOPT_CAINFO, m_oc.options["ca_bundle"].c_str());
-			}else{
-				res = curl_easy_setopt(curl, CURLOPT_CAPATH, m_oc.options["ca_path"].c_str());
+				res = curl_easy_setopt(m_curl, CURLOPT_CAPATH, m_oc.options["ca_path"].c_str());
 			}
 		}
 
-  		if(res == CURLE_OK)
+		if(m_oc.options["echo"] == std::string("false"))
 		{
-			res = curl_easy_perform(curl);
+			// If unset, libcurl defaults to fwrite to stdout, ie: echoing
+			if(res == CURLE_OK)
+			{
+				res = curl_easy_setopt(m_curl, CURLOPT_WRITEFUNCTION, noop_write_callback);
+			}
 		}
+	}
 
-		if(res != CURLE_OK)
-		{
-			falco_logger::log(LOG_ERR, "libcurl error: " + std::string(curl_easy_strerror(res)));
-		}
-		curl_easy_cleanup(curl);
-		curl = NULL;
-		curl_slist_free_all(slist1);
-		slist1 = NULL;
+	if(res != CURLE_OK)
+	{
+		err = "libcurl error: " + std::string(curl_easy_strerror(res));
+		return false;
 	}
+	return true;
+}
+
+void falco::outputs::output_http::output(const message *msg)
+{
+	CURLcode res = curl_easy_setopt(m_curl, CURLOPT_POSTFIELDS, msg->msg.c_str());
+	if (res == CURLE_OK)
+	{
+		res = curl_easy_perform(m_curl);
+	}
```

Comment:
Here is not so relevant but in the `init` function it could cleanup a little bit
```suggestion
	res = res == CURLE_OK? curl_easy_perform(m_curl) : res;
```